mitiq/observable/pauli.py

```diff
@@ -33,6 +33,9 @@
 
 
 class PauliString:
+    """A sequence of Pauli operators defined on the space spanned by the qubits
+     in the qubit register of the quantum circuit."""
```

Comment:
I think its misleading/confusing to say this is defined in terms of qubits in a circuit. A `PauliString` is a (tensor) product of single-qubit Paulis I, X, Y, and Z with a leading (real or complex) coefficient. `PauliString`s can be measured in any `mitiq.QPROGRAM`, irrespective of the qubits or qubit type.